marklogic-client-api/src/main/java/com/marklogic/client/example/cookbook/datamovement/BulkExportWithDataService.java

```diff
@@ -0,0 +1,133 @@
+import com.marklogic.client.datamovement.QueryBatch;
+import com.marklogic.client.datamovement.QueryBatchListener;
+import com.marklogic.client.datamovement.QueryBatcher;
+import com.marklogic.client.document.DocumentWriteSet;
+import com.marklogic.client.document.JSONDocumentManager;
+import com.marklogic.client.document.TextDocumentManager;
+import com.marklogic.client.io.DocumentMetadataHandle;
+import com.marklogic.client.io.StringHandle;
+import com.marklogic.client.example.cookbook.Util;
+import com.marklogic.client.query.*;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+
+public class BulkExportWithDataService implements QueryBatchListener {
+
+    private DatabaseClient dbClient = DatabaseClientSingleton.getAdmin("java-unittest");
+    private DatabaseClient dbModulesClient = DatabaseClientSingleton.getAdmin("java-unittest-modules");
+    private DataMovementManager moveMgr = dbClient.newDataMovementManager();
+    private List<String> urisList = new ArrayList<>();
+
+    @Override
+    public void processEvent(QueryBatch batch) {
+    }
+
+    public static void main(String args[]) throws IOException {
+        new BulkExportWithDataService().run();
+    }
+
+    private void run() throws IOException {
+        setup();
+        exportJson();
+        tearDown();
+    }
+
+    private void setup() throws IOException {
+
+        writeDocuments(100,"BulkExportWithDataService");
+        writeScriptFile("readJsonDocs.api");
+        writeScriptFile("readJsonDocs.sjs");
+    }
+
+    private  void tearDown(){
+
+        QueryManager queryMgr = dbClient.newQueryManager();
+        QueryManager queryMgrModules = dbModulesClient.newQueryManager();
+        DeleteQueryDefinition deletedef = queryMgr.newDeleteDefinition();
+        deletedef.setCollections("BulkExportWithDataService");
+        queryMgr.delete(deletedef);
+        queryMgrModules.delete(deletedef);
+    }
+
+
+    private void exportJson(){
+        BulkExportServices bulkExportServices = uris -> {
+            List<Object> list = Arrays.asList(uris.toArray());
+            for(Object i:list) {
+                if(!(urisList.contains(i)))
+                    throw new InternalError("urisList does not contain "+i.toString());
+                urisList.remove(i);
+            }
+            return null;
+        };
+
+        StructuredQueryBuilder structuredQueryBuilder = new StructuredQueryBuilder();
+        structuredQueryBuilder.directory(1,"/example/cookbook/bulkExport/");
+        QueryBatcher queryBatcher = moveMgr.newQueryBatcher(structuredQueryBuilder.collection("BulkExportWithDataService"))
+                .withBatchSize(3)
+                .withThreadCount(3)
+                .onQueryFailure(batch -> new InternalError("An exception occured in queryBatcher"))
+                .onUrisReady(batch -> bulkExportServices.readJsonDocs(Stream.of(batch.getItems())));
```

Comment:
readJsonDocs() returns a stream of JSON documents.  Would it make sense to log those documents inside the lambda function by calling forEach() on the stream of JSON documents with a lambda that writes the Reader for each document to System.out?